src/main/scala/com/yahoo/tensorflowonspark/TFModel.scala

```diff
@@ -0,0 +1,340 @@
+    * @param tensors Sequence of M TensorFlow Tensors, each with N items in the 0-th dimension.
+    * @return List of N Rows with M Columns.
+    */
+  def tensors2batch(tensors: Seq[Tensor[_]]): List[Row] = {
+    // all output tensors must have same cardinality in the 0-dimension
+    assert(tensors.map(t => t.shape()(0)).distinct.size == 1)
+    val numRows = tensors.head.shape()(0).toInt
+
+    // convert Tensors to list of (type, shape, flattened nio buffers)
+    var tensorArrays = ListBuffer.empty[(DataType, Array[Long], Buffer)]
+    for (t <- tensors) {
+      val dtype = t.dataType()
+      dtype match {
+        case DataType.BOOL =>
+          val buf = ByteBuffer.allocate(t.numElements)
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.DOUBLE =>
+          val buf = DoubleBuffer.allocate(t.numElements)
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.FLOAT =>
+          val buf = FloatBuffer.allocate(t.numElements)
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.INT32 =>
+          val buf = IntBuffer.allocate(t.numElements)
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.INT64 =>
+          val buf = LongBuffer.allocate(t.numElements)
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.STRING =>
+          val buf = ByteBuffer.allocate(t.numBytes())
+          t.writeTo(buf)
+          buf.flip()
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case DataType.UINT8 =>
+          val buf = ByteBuffer.allocate(t.numBytes())
+          t.writeTo(buf)
+          tensorArrays = tensorArrays :+ (dtype, t.shape, buf)
+        case unsupportedType =>
+          throw new Exception(s"Unsupported output Tensor type: $unsupportedType")
+      }
+    }
+
+    var result = ListBuffer.empty[Row]
+    for (i <- 0 until numRows) {
+      val slices = tensorArrays.map { case (dtype, shape, buf) =>
+        // 0-th dimension should be the number of rows
+        if (shape.length == 1) {
+          // scalar values
+          dtype match {
+            case DataType.BOOL =>
+              buf.asInstanceOf[ByteBuffer].get() == 1.toByte
+            case DataType.DOUBLE =>
+              buf.asInstanceOf[DoubleBuffer].get()
+            case DataType.FLOAT =>
+              buf.asInstanceOf[FloatBuffer].get()
+            case DataType.INT32 =>
+              buf.asInstanceOf[IntBuffer].get()
+            case DataType.INT64 =>
+              buf.asInstanceOf[LongBuffer].get()
+            case DataType.STRING =>
+              buf.asInstanceOf[ByteBuffer].get(Array.ofDim[Byte](buf.remaining()))
```

Comment:
Does this put entire tensor into first row?   Same for UINT8.   Appears there is no difference between this and array case below.

Reply:
Yes, Binary and String types are "special" in that they represent "scalar" types with an array of bytes.  And, I went ahead and left them to be "separate but identical" for now since it's easier to read, and just in case they might diverge in the future.